provider/provider.go

```diff
@@ -29,6 +29,22 @@ import (
 type Provider interface {
 	Records(ctx context.Context) ([]*endpoint.Endpoint, error)
 	ApplyChanges(ctx context.Context, changes *plan.Changes) error
+	AttributeValuesEqual(attribute string, value1 *string, value2 *string) bool
+}
+
+type BaseProvider struct {
+}
+
+func (b BaseProvider) AttributeValuesEqual(attribute string, value1 *string, value2 *string) bool {
+	if (value1 == nil) != (value2 == nil) {
+		return false
+	}
+
+	if value1 == nil {
+		return true
+	}
```

Comment:
The two ifs above seem like breaking change

Reply:
ok I understand that they can be nil only for custom attributes when they aren't set. I think the better default would be to set value1 and value2 to empty strings. this way there is no need to handle these special cases, receive *string instead of string, and the case then value of custom attribute is set to empty string is handled by default

Reply:
I had thought that there might conceivably be cases where there was a difference between not-set and empty string. I didn't have anything in particular in mind (I don't *know* of such a case OTOH), but it seems not unlikely that it could happen.

Reply:
> The two ifs above seem like breaking change

As you say, they're only there to deal with customer attributes that aren't set. So, just to be clear, they *aren't* a breaking change.